datumaro/components/algorithms/noisy_label_detection/loss_dynamics_analyzer.py

```diff
@@ -44,28 +44,46 @@ class LossDynamicsAnalyzer:
     International Conference on Learning Representations. 2021.
     """
 
-    allowed_task_names = {"OTX-MultiClassCls"}
+    allowed_task_names = {"OTX-MultiClassCls", "OTX-Det"}
+    allowed_tracking_loss_types = {"cls", "bbox", "centerness", "bbox_refine"}
 
-    def __init__(self, dataset: IDataset, alpha: float = 0.001) -> None:
+    def __init__(
+        self, dataset: IDataset, alpha: float = 0.001, tracking_loss_type: Optional[str] = None
+    ) -> None:
         purpose = dataset.infos().get("purpose")
         if purpose != "noisy_label_detection":
             raise DatasetError(
                 f'Dataset infos should have purpose="noisy_label_detection", but it has {purpose}.'
             )
 
         task = dataset.infos().get("task")
-        if task != "OTX-MultiClassCls":
+        if task not in self.allowed_task_names:
             raise DatasetError(
-                f"{task} is not allowed. The allowed task names are {self.allowed_task_names}."
+                f"task={task} is not allowed. The allowed task names are {self.allowed_task_names}."
+            )
+
+        if task == "OTX-Det" and tracking_loss_type is None:
+            raise DatasetError(
+                'If task="OTX-Det", tracking_loss_type should not be None. '
+                f"The allowed tracking loss types are {self.allowed_tracking_loss_types}."
+            )
+        elif (
```

Comment:
```suggestion
        if (
```

Reply:
You might achieve consistency between if clauses L60, L65, and L70. I guess that we can achieve it by introducing `elif` to L65 as well. This is because I want to L60-L76 block to be seen as the one who check the input argument.

Reply:
Both look good to me. thank you for checking.